### PR TITLE
Enable cross-building for Scala 2.11 and 2.12

### PR DESCRIPTION
**Pull request by wjoel**

### CODE REVIEW

File: src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala [hunk 1]
```diff
@@ -47,10 +47,11 @@ class AssetsRelation(config: RelationConfig, assetPath: Option[String])(val sqlC
     MetricsSource.getOrCreateCounter(config.metricsPrefix, s"assets.created")
 
   override def insert(df: org.apache.spark.sql.DataFrame, overwrite: scala.Boolean): scala.Unit =
-    df.foreachPartition(rows => {
+    df.foreachPartition((rows: Iterator[Row]) => {
```
Comment (wjoel) [inline src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: Scala 2.12 needs some help to get the right overloaded method of `foreachPartition`.

File: src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala [hunk 2]
```diff
@@ -47,10 +47,11 @@ class AssetsRelation(config: RelationConfig, assetPath: Option[String])(val sqlC
...
       implicit val contextShift: ContextShift[IO] = IO.contextShift(ExecutionContext.global)
       val batches = rows.grouped(config.batchSize.getOrElse(Constants.DefaultBatchSize)).toVector
       batches.parTraverse(postRows).unsafeRunSync()
+      ()
```
Comment (hakontro) [inline src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: This is unusual, perhaps we should add a comment to explain why it's needed?

Comment (Tapped) [reply src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: Hmm, that's the only way I know for how to create an unit type. Though `unsafeRunRync()` should return an unit in this case, I believe.

Comment (wjoel) [reply src/main/scala/com/cognite/spark/datasource/AssetsRelation.scala]: It's part of the "Scala 2.12 needs some help to get the right overloaded method of `foreachPartition`", without this the return type is `Vector[Unit]` and the compiler doesn't know what to do.

`()` explicitly makes it return the required `Unit` type. I don't want to add comments on this everywhere it's done.

**Pull request merged**